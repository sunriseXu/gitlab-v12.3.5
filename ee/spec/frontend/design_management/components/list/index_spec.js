import { shallowMount } from '@vue/test-utils';
import List from 'ee/design_management/components/list/index.vue';

const createMockDesign = id => ({
  id,
  filename: 'test',
  image: 'test',
  event: 'NONE',
  notesCount: 2,
  updatedAt: '01-01-2019',
});

describe('Design management list component', () => {
  let wrapper;

  function createComponent() {
    wrapper = shallowMount(List, {
      sync: false,
      propsData: {
        designs: [createMockDesign(1), createMockDesign(2)],
      },
    });
  }

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders list', () => {
    createComponent();

    expect(wrapper.element).toMatchSnapshot();
  });
});
