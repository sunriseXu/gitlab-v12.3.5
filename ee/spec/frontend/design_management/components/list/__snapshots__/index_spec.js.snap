// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design management list component renders list 1`] = `
<ol
  class="list-unstyled row"
>
  <li
    class="col-md-6 col-lg-4 mb-3"
  >
    <design-stub
      event="NONE"
      id="1"
      image="test"
      name="test"
      notescount="2"
      updatedat="01-01-2019"
    />
  </li>
  <li
    class="col-md-6 col-lg-4 mb-3"
  >
    <design-stub
      event="NONE"
      id="2"
      image="test"
      name="test"
      notescount="2"
      updatedat="01-01-2019"
    />
  </li>
</ol>
`;
